Review of the Getting Started content (#404)
neo4j/docs-getting-started: modules/ROOT/pages/graph-database.adoc

:page-ad-overline: Neo4j GraphAcademy
:page-ad-title: Neo4j Fundamentals
:page-ad-description: Learn the basics of Neo4j and the property graph model
:page-ad-link: https://graphacademy.neo4j.com/courses/neo4j-fundamentals/?ref=docs
:page-ad-underline-role: button
:page-ad-underline: Learn more

A Neo4j graph database stores data as <<nodes>>, <<relationships>>, and <<properties>> instead of tables or documents.
This means you can organize your data in a similar way as when sketching ideas on a whiteboard.

And since graph databases are not restricted to a pre-defined model, you can take more flexible approaches and strategies when working with them.

.A visual way of matching patterns and relationships using Cypher
image::ROOT:sample-cypher.svg[]

== How it works

Graph databases are structured through nodes and relationships.

*Nodes* are entities in the graph which can:

* Be tagged with <<labels>> representing their different roles in a domain (e.g., `Person`).
* Hold any number of link:https://neo4j.com/docs/cypher-manual/current/values-and-types/[key-value pairs] and properties (e.g., `name`).

> * Include metadata such as <<indexes>> or <<constraints>> information.

Reviewer: It's confusing there as well. Let's see what Jens can make of it!

*Relationships* provide named connections between two node entities (e.g., _Person_ - `LOVES` - _Person_) and they:

* Must always have a start node, an end node, and a xref:https://neo4j.com/docs/cypher-manual/current/values-and-types/[type] (e.g. )
* Can be directed or undirected.
* Can have properties, like nodes.

> * Can be attached to nodes, no matter how many they are and of what type, without sacrificing performance.

Reviewer: The original sentence is probably simpler.

* Are very useful as a strategy for xref:data-modeling/index.adoc[modeling your data].

In summary, nodes and relationships are an efficient and flexible approach because, with them, you can:

* Create constant time traversals in big graphs for both depth and breadth.

> * Scale-up your database to billions of nodes while using moderate hardware.

Reviewer: I was referring to "hardware". I think the original sentence is quite old..

> * Scale-up your database to billions of nodes while using moderate hardware.

Author: Ah yes, so maybe specifying it's about self-managed deployments or remove the whole sentence? If we keep it, it could be something like

`Scale-up your database to billions of nodes while using moderate hardware on local deployments.`

* Design flexible property graph <<schemas>> that can adapt over time.

[[why-graphdb]]

> == Why use it

Reviewer: The title should do away with any confusion about what it is.

Often times, projects can deal with large amounts of data that happens to be also very complex.
In such cases, graph databases come to be a powerful tool.

There are different kinds of databases, such as relational databases which can also store relationships.
However, relational databases use computing-wise expensive `JOIN` operations or cross-looks, which are often tied to a rigit schema.

Graph databases do not use these methods.

> Relationships are stored natively alongside the data elements (nodes) in a more flexible format, which allows the optimization of data traversing and millions of connections to be accessed per second, per core.

Reviewer: Right, but regardless of how many cores you are using, a join is always more expensive than traversing a graph so it feels a bit out of place to start talking about cores here.

Moreover, many daily challenges and tasks can be viewed from a graph perspective as it allows you to:

* Navigate deep hierarchies.
* Find hidden connections between distant items.
* Discover inter-relationships between items.

== How to use

image::use-case-summary.svg[role="popup-link"]

Whether it's a social or a road network, everything can be structured as an interconnected graph of relationships. 
Many times, any project's questions and challenges revolve around the relationship between elements, not the elements themselves (e.g. how to get from A to B, instead of what is A and what is B).
For this reason, graphs can be applied to many areas of society and to a wide variety of projects.

Neo4j is used today by thousands of https://neo4j.com/customers/[startups, educational institutions, and large enterprises] in all sectors including financial services, government, energy, technology, retail, and manufacturing.
Graphs have been successful in helping them on the development of innovative new technology, business management, insight and revenue regenration, as well as overall improvements in efficiency.

> You can find more information link:https://neo4j.com/use-cases/[on use cases] by navigating through the main website.

Reviewer: ```suggestion
You can find more information about use cases on link:https://neo4j.com/use-cases/[Neo4j's main website].
```

== Keep learning

> If you want to have a better and deeper understanding of what a graph database is, read more about xref:reference/graphdb-concepts/index.adoc[Graph database concepts], and enroll to the GraphAcademy course on link:https://graphacademy.neo4j.com/courses/neo4j-fundamentals/[Neo4j Fundamentals].

Reviewer: ```suggestion
If you want to have a better and deeper understanding of graph databases, you can read more about xref:reference/graphdb-concepts/index.adoc[Graph database concepts], or enroll to the GraphAcademy course on link:https://graphacademy.neo4j.com/courses/neo4j-fundamentals/[Neo4j Fundamentals].
```
